lab-taylor/controller/team-controller.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const debug = require('debug')('authlab:team-controller');
+const httpErrors = require('http-errors');
+const Team = require('../model/team');
+
+exports.createTeam = function(teamData) {
+  debug('creating team');
+  return new Promise((resolve, reject) => {
+    new Team(teamData).save()
+    .then(team => resolve(team))
+    .catch(err => reject(httpErrors(400, err.message)));
+  });
+};
+
+exports.fetchTeam = function(teamId) {
+  debug('fetching teams', teamId);
+  return new Promise((resolve, reject) => {
+    Team.findOne({_id: teamId})
+    .then(team => {
+      resolve(team);
+    })
+    .catch(() => reject(httpErrors(404, 'team not found')));
+  });
+};
+
+exports.updateTeam = function(teamId, reqBody) {
+  debug('updating item', teamId);
+  debug('team body', reqBody);
+  return new Promise((resolve, reject) => {
+    if(Object.keys(reqBody).length === 0) return reject(httpErrors(400, 'need to provide a body'));
+
+    const teamKeys = ['name', 'city', 'coach'];
+    Object.keys(reqBody).forEach((key) => {
+      if (teamKeys.indexOf(key) === -1) return reject(httpErrors(400, 'key does not exist'));
+    });
+
+    Team.findByIdAndUpdate(teamId, reqBody)
```

Comment:
I'm assuming you are looking up the team again to get its updates. To make this work the way you want, add an option to the arguments:

Team.findByIdAndUpdate(teamID, reqBody, {new: true})
.then( team => // now the passed in "team" is your updated team :) );